css/apigee_edge.app_view.css

```diff
@@ -15,13 +15,13 @@
  * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  */
 
-.field {
+[class*="-app--view-mode-full"] .field {
```

Comment:
Wouldn't be better to add a common class to developer and teams apps? E.g. `apigee-entity--app` or something like that. That way we can avoid always using attribute selectors in the future.

Reply:
@mxr576 We are doing something similar in Kickstart. See https://github.com/apigee/apigee-devportal-kickstart-drupal/blob/8.x-1.x/modules/custom/apigee_kickstart_enhancement/templates/app.html.twig

Reply:
I did not want to create a dedicated app (developer/team app) theme and template in the module, I wanted to keep it simple. Every Apigee entity uses the same default Apigee entity template and if you would like to use a custom template, you just need to define a theme function with the name of the entity type - as the header in the apigee-entity-default.html.twig describes.

The header also describes that it is an intended behavior that there are entity type specific template suggestions for the apigee_entity_default theme. 

https://github.com/apigee/apigee-edge-drupal/blob/02a29d6f5082306bfa319eca57c5c157962ead1a/templates/apigee-entity-default.html.twig#L9-L31

Thanks for this approach we need to worry less about BC breaking issues in the theme layer. If someone starts customizing the Apigee entity view pages it starts with a clean sheet and it maintains its own customization.

Reply:
@arshad What you referenced from the kickstarter_enhancements module is what I am actually trying to achieve here. Modules and themes that would like to customize a display of an Apigee Edge entity should implement and maintain its own theme.

Reply:
I didn't say that you should create dedicated templates, just add a common class to both entities like `apigee-entity--app` to make frontenders' life easier :)

Reply:
Ohh, I got it now. :) If there is an only reason to define an "app" theme and force it to be applied on developer- and team apps I would say it adds too much complexity to the code compared with the value it gives. I mean, you can theme the app page by using the default selectors provided by the apigee_edge_default theme with those selectors that I used in the app_view.css. Yes, `apigee-entity--app` would look nicer but we would need to maintain a new theme hook, preprocess function and template just to add this "nice" class to apps.
We are not trying to provide a complex front-end in this module, just something that works. Front-enders will customize it anyway according to the needs of a customer. This is the reason why we are trying to use Drupal core render elements only. I think this rule can be also applied here, we should not introduce a template for apps unless we have a really good reason.

Reply:
I see two ways to do that without creating a dedicated developer_app/team_app theme:

1. Create a common `app` theme and apply it on developer_app and team_apps in its view builder, as the apigee_kickstarter_enhancements does. I would not add additional complexity like this to the code just to add one "beautiful" CSS class to a render element. If the "apigee_entity" theme would not define any selector that front-enders could use by default that would be different situation. But this is not the case.
2. Attaching this custom CSS class t app entities in the `template_preprocess_apigee_entity()` function based on the type of the entity. I would not do this, because I want to keep the "apigee_entity" theme generic, an actual fallback/workaround for the Drupal core issue that it addresses. But you will be able to add this nice class to apps in your module's `MY_MODULE_preprocess_apigee_entity()` function unless you want to define the "app" theme and override the entity view builder on apps.

Reply:
The updated PR still does not add this "pretty" class to apps because of the reasons that I described in my previous comment ^.

Reply:
(Another option that came in my mind is allowing to define additional classes in the entity type annotation, but it seems an overkill just to tag all kinds of apps with a pretty "app" class.)

Reply:
Thanks for the reasoning. We can add the class from the preprocess function you mentioned.